agent/agent.go

```diff
@@ -2231,10 +2242,21 @@ func (a *Agent) loadChecks(conf *config.RuntimeConfig) error {
 			return fmt.Errorf("failed reading check file %q: %s", file, err)
 		}
 
+		// If the file ended up empty as a result of something like an OS crash, remove
+		// it for convenience and log about it.
+		if len(buf) == 0 {
+			a.logger.Printf("[WARN] Removing leftover empty check file %q", file)
+			if err := os.Remove(file); err != nil {
+				a.logger.Printf("[WARN] Error removing leftover empty check file %q: %v", file, err)
+			}
+			continue
+		}
+
 		// Decode the check
 		var p persistedCheck
 		if err := json.Unmarshal(buf, &p); err != nil {
-			return fmt.Errorf("Failed decoding check file %q: %s", file, err)
+			a.logger.Printf("[WARN] Failed decoding check file %q: %s", file, err)
```

Comment:
I was torn on this one, but since these are internal to Consul then we ought to skip these if we skip empty files, since OS crashes could cause this, too. I'd change all the logging in here to `ERR` level though.